client/src/components/restaurant/restaurant.js

```diff
@@ -0,0 +1,65 @@
+import React, { Component } from 'react';
+import { Tab, Tabs, Container } from 'react-bootstrap';
+import PropTypes from 'prop-types';
+import Api from '../../services/api';
+import About from './about';
+import '../../style/restaurant.css';
+
+class Restaurant extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      restaurant: {},
+      isFetching: false,
+    };
+  }
+
+  async componentDidMount() {
+    this.getOne();
+  }
+
+  getOne() {
+    const { match } = this.props;
+    Api.getOne('restaurants', match.params.id)
+      .then((response) => {
+        if (response.error) {
+          // eslint-disable-next-line no-console
+          console.error(response);
+          return;
+        }
+        this.setState({
+          restaurant: response.data,
+          isFetching: true,
+        });
+      });
+  }
```

Comment:
I think it is better to try to `setState()` before checking if response has got any error. I would switch lines 25 and 30 placing. 

Of course, we should check first `if(!response.error)` and only after this `else`.

Reply:
I set `isFetching = false` in constructor so nested component won't break. If request returns error than `response.data = undefined` and we don't need to update state before if statement

Reply:
I like your way. It seems good.

On the same time I guess we will have more successful

Reply:
I like your way. It seems good.

At the same time I guess we'll have more successful responses rather than failed. That's why I would prefer to check ones but not twice.

Reply:
In this case
```
then((response) => {
        this.setState({
          restaurant: response.data,
          isFetching: true,
        });
        if (response.error) {
          // eslint-disable-next-line no-console
          console.error(response);
          this.setState({
            restaurant: {},
            isFetching: false,
          });
          // return;
        }
      });
```
I have this result if error happens
![image](https://user-images.githubusercontent.com/6152066/76287387-0a4e7680-62ad-11ea-9031-ee61f7fc822c.png)

I use Error-First principle to handle errors http://fredkschott.com/post/2014/03/understanding-error-first-callbacks-in-node-js/
It doesn't affected user experience because if statement works very fast

But anyway, could you show code example with your solution?

Reply:
```
      .then((response) => {
        if (!response.error) {
            this.setState({
                restaurant: response.data,
                isFetching: true,
            });
            return;
        } else {
            console.error(response);
        }
      });
  }
```

Reply:
I try to avoid negative conditionals https://gist.github.com/wojteklu/73c6914cc446146b8b533c0988cf8d29#understandability-tips

It makes code less readable